### ISSUE TITLE
Parser includes trailing \n in scalar if next line has whitespace

### ISSUE THREAD

**Issue by Gei0r** [biojppm/rapidyaml#153]
An unquoted scalar is parsed with a trailing newline if the following line contains only spaces.

```c++
#include <ryml.hpp>
#include <iostream>

int main()
{
    ryml::Tree tr = ryml::parse("- A\n \n");   // same problem with: "a: A\n \n"
    std::cout << '>' << tr[0].val() << "<\n";
}
```
```
Expected output: >A<

Actual output: >A
<

(value has a trailing newline)
```

**Comment by biojppm** [comment]
Thanks for reporting.

**Comment by biojppm** [comment]
@Gei0r failed to reproduce this. Tried the following, and it succeeds:
```c++
TEST(plain_scalar, issue153_seq)
{
    Tree t = parse("- A\n \n");
    EXPECT_EQ(t[0].val(), "A");
}

TEST(plain_scalar, issue153_map)
{
    Tree t = parse("foo: A\n \n");
    EXPECT_EQ(t["foo"].val(), "A");
}
```

**Comment by Gei0r** [comment]
I bisected it and the problem was fixed in b5e1bdd45d110da0d8036fa1647f90f09680146f 26 days ago (10 days before my issue), so I guess I didn't have the latest version.

**Comment by biojppm** [comment]
Maybe that was merged after. And sorry about me forgetting that fix; a lot of stuff was fixed around that time.

Anyway, thanks again.
